cpp/log/cluster_state_controller-inl.h

```diff
@@ -0,0 +1,203 @@
+#ifndef CERT_TRANS_LOG_CLUSTER_STATE_CONTROLLER_INL_H_
+#define CERT_TRANS_LOG_CLUSTER_STATE_CONTROLLER_INL_H_
+
+#include <functional>
+
+#include "log/cluster_state_controller.h"
```

Comment:
Still sort of wrong, I'd say... This header should be the very first one, to ensure that it's not "fixed" by ```#include <functional>```.